code/modules/station_goals/bluespace_tap.dm

```diff
@@ -455,20 +427,88 @@
 /**
   * Produces the product with the desired key and increases product cost accordingly
   */
-/obj/machinery/power/bluespace_tap/proc/produce(key)
-	if(key <= 0 || key > length(product_list))	//invalid key
-		return
-	var/datum/data/bluespace_tap_product/A = product_list[key]
-	if(!A)
-		return
-	if(A.product_cost > points)
+/obj/machinery/power/bluespace_tap/proc/produce(var/datum/data/bluespace_tap_product/product, purchased = FALSE, double_chance = FALSE)
+	if(!product)
 		return
-	points -= A.product_cost
-	A.product_cost = round(1.2 * A.product_cost, 1)
+	if(purchased)
+		if(product.product_cost > points)
+			return
+		points -= product.product_cost
+		product.product_cost = round(1.2 * product.product_cost, 1)
 	playsound(src, 'sound/magic/blink.ogg', 50)
 	do_sparks(2, FALSE, src)
-	new A.product_path(get_turf(src))
+
+	var/spawn_location = find_spawn_location()
+	spawn_item(product, spawn_location)
+
+	if(prob(25) && double_chance)
+		// Spawn second item in random spot on station - places it where NADs can respawn
+		spawn_location = find_spawn_location(TRUE)
+		spawn_item(product, spawn_location)
+
+/// Handles a motherlode - each product is spawned 5 times at both the machine and around the station
+/obj/machinery/power/bluespace_tap/proc/produce_motherlode()
+	// Announce lootsplosion
+	GLOB.major_announcement.Announce("Power spike detected during Bluespace Harvester Operation. Large bluespace payload inbound.", "Bluespace Harvester Motherlode", 'sound/AI/attention.ogg')
+	// Spawn lootsplosion
+	for(var/datum/data/bluespace_tap_product/product in product_list)
+		for(var/i in 1 to 5)
+			var/spawn_location = find_spawn_location()
+			spawn_item(product, spawn_location)
+			spawn_location = find_spawn_location(TRUE)
+			spawn_item(product, spawn_location)
+
+/obj/machinery/power/bluespace_tap/proc/find_spawn_location(random = FALSE)
+	var/list/possible_spawns = list()
+	if(random)
+		possible_spawns = GLOB.nukedisc_respawn
```

Comment:
As a maintainer, I do not think nuke disc respawn should not be used for anything else codewise. If you do want to use this, please make it a random turf AROUND the nuke disc respawn (1-2 tile range) so you dont get 10 items on the same tile.